core/types/eth_transaction.go

```diff
@@ -177,6 +176,29 @@ func (tx *EthTransaction) Copy() *EthTransaction {
 	return &tx2
 }
 
+// ConvertToHmy converts eth txn to hmy txn by filling in ShardID and ToShardID fields.
+func (tx *EthTransaction) ConvertToHmy() *Transaction {
+	var tx2 Transaction
+	d := &tx.data
+	d2 := &tx2.data
+
+	d2.AccountNonce = d.AccountNonce
+	d2.Price = new(big.Int).Set(d.Price)
+	d2.GasLimit = d.GasLimit
+	d2.Recipient = copyAddr(d.Recipient)
+	d2.Amount = new(big.Int).Set(d.Amount)
+	d2.Payload = append(d.Payload[:0:0], d.Payload...)
+	d2.V = new(big.Int).Set(d.V)
+	d2.R = new(big.Int).Set(d.R)
+	d2.S = new(big.Int).Set(d.S)
+	d2.Hash = copyHash(d.Hash)
```

Comment:
Seems like it's better to recompute the hash once done setting the hmy tx fields. Since Tx hash is a hash of the RLP encoding of the tx obj:

https://github.com/harmony-one/harmony/blob/9381b95cdc9ba402773002e2a79eee91a1269c7c/core/types/transaction.go#L392-L401

Reply:
Good catch. done.